umbilical/x509.go

```diff
@@ -83,3 +85,132 @@ func GetChainFromTLSServer(addr string) (chain []*x509.Certificate, err error) {
 	wg.Wait()
 	return
 }
+
+// Checks whether the given assertion (to be) issued in the given batch
+// is consistent with the given X.509 certificate chain and
+// trusted roots.
+//
+// We are more strict than is perhaps required. For instance, we do
+// not allow an assertion for some.example.com to be backed
+// by a wildcard certificate for *.example.com.
+// Also we require basically the same chain to be valid for the full
+// duration of the assertion.
+//
+// If consistent, returns one or more verified chains. This is useful
+// for revocation checks.
+//
+// Note: does not perform any revocation check. Also does not check SCTs.
+func CheckAssertionValidForX509(a mtc.Assertion, batch mtc.Batch,
+	chain []*x509.Certificate, roots *x509.CertPool) (
+	[][]*x509.Certificate, error) {
+	if len(chain) == 0 {
+		return nil, errors.New("empty chain")
+	}
+
+	cert := chain[0]
+
+	// Check if claims match certificate.
```

Comment:
Note: this is really checking that the claim is covered by the certificate, but not that they're an exact match. If there are extra IPs in the certificate that aren't included in the claim, that'll pass this check. Is that what we want? I think yes, but let's clarify the comment.

Reply:
Elaborated.